From 09e320bb69826db674f4adf9fd9acf20005211c0 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:15:32 -0400
Subject: [alsa] oxygen: Xonar DG(X) capture from I2S channel 1, not 2

Message-id: <140972823573-59-patchset-jkysela@redhat.com>
Patchwork-id: 90759
O-Subject: [RHEL 7.1 PATCH 058/442] ALSA: oxygen: Xonar DG(X): capture from I2S channel 1, not 2
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Actually CS4245 connected to the I2S channel 1 for
capture, not channel 2. Otherwise capturing and
playback does not work for CS4245.

Signed-off-by: Roman Volkov <v1ron@mail.ru>
Signed-off-by: Clemens Ladisch <clemens@ladisch.de>

(picked from kernel: 3dd77654fb1d7f68b9739f3039bad8dbbc0739f8)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/oxygen/xonar_dg_mixer.c b/sound/pci/oxygen/xonar_dg_mixer.c
index a2cd0d3..7e9fc4a 100644
--- a/sound/pci/oxygen/xonar_dg_mixer.c
+++ b/sound/pci/oxygen/xonar_dg_mixer.c
@@ -369,7 +369,7 @@ struct oxygen_model model_xonar_dg = {
  .model_data_size = sizeof(struct dg),
  .device_config = PLAYBACK_0_TO_I2S |
     PLAYBACK_1_TO_SPDIF |
-    CAPTURE_0_FROM_I2S_2 |
+    CAPTURE_0_FROM_I2S_1 |
     CAPTURE_1_FROM_SPDIF,
  .dac_channels_pcm = 6,
  .dac_channels_mixer = 0,
-- 
1.7.1